File: 647-palindromic-substrings/palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        result = 0
        for i in range(len(s)):
            l = r = i
            while -1 < l and r < len(s) and s[l] == s[r]:
                l -= 1
                r += 1
                result += 1
            l = i
            r = i + 1
            while -1 < l and r < len(s) and s[l] == s[r]:
                l -= 1
                r += 1
                result += 1
        return result

File: 647-palindromic-substrings/palindromic-substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int result = 0;
        for (int i = 0; i < s.length(); i++){
            int l = i, r = i;
            while ( -1 < l && r < s.length() && s[l] == s[r] ){
                result++;
                l--;
                r++;
            }
            l = i;
            r = i + 1;
            while ( -1 < l && r < s.length() && s[l] == s[r] ){
                result++;
                l--;
                r++;
            }
        }
        return result;
    }
};